uiTests/src/test/java/com/virtuslab/gitmachete/uitest/UiTestSuite.java

```diff
@@ -60,16 +70,33 @@ public void openTabAndCountRows() {
     Assert.assertEquals(13, branchAndCommitRowsCount);
   }
 
+  @Test
+  public void pullCurrentBranchAndEnsureCleanWorkingTree() {
+    runJs("ide.soleOpenedProject().openTab()");
+
+    runJs("ide.soleOpenedProject().checkoutBranch('allow-ownership-link')");
+    runJs("ide.soleOpenedProject().pullBranch('allow-ownership-link')");
+
+    ArrayList<String> changes = callJs("ide.soleOpenedProject().getDiffOfWorkingTreeToHead()");
+    Assert.assertEquals(new ArrayList<>(), changes);
```

Comment:
**TODO: This is now failing.**

Reply:
`    java.lang.AssertionError: expected:<[]> but was:<[D: /tmp/machete-tests-10763193968752472041/machete-sandbox/allow-ownership-link-1st-round-of-fixes.txt]>`

Reply:
Ok, fixed by PR #438